testsupport/wait/awaitility.go

```diff
@@ -458,6 +460,19 @@ func (a *Awaitility) WaitForDeploymentToGetReady(name string, replicas int) {
 		if int(deployment.Status.AvailableReplicas) != replicas {
 			return false, nil
 		}
+		pods := &corev1.PodList{}
+		require.NoError(a.T, a.Client.List(context.TODO(), pods, client.InNamespace(a.Namespace), client.MatchingLabels(deployment.Spec.Selector.MatchLabels)))
+		if len(pods.Items) != replicas {
+			return false, nil
+		}
+		for i := range pods.Items {
+			if util.IsBeingDeleted(&pods.Items[i]) {
+				return false, nil
+			}
+			if !podutils.IsPodReady(&pods.Items[i]) {
+				return false, nil
+			}
+		}
```

Comment:
```suggestion
		for _, pod := range pods.Items {
			if util.IsBeingDeleted(&pod) || !podutils.IsPodReady(&pod) {
				return false, nil
			}
		}
```

Reply:
yeah, why not - just the linter will fail with this change because of the implicit memory aliasing in for loop

Reply:
ah yes, probably 🤦‍♂️ 
but at least the 2 "if" blocks can be combined, can't they? anyways, I'll leave that to you ;)

Reply:
yes, I noticed the if statement change - thanks for it - I had already applied that, and then I updated it so the linter is ignored ;-)